modules/optimization/doc/content/modules/optimization/examples/materialInv_diffusion_reaction.md

```diff
@@ -0,0 +1,42 @@
+# Material Inversion Example: Nonlinear Diffusion Reaction
+
+The following illustrates the capability of the MOOSE-Opt by applying the module to a nonlinear, material-inversion optimization problem:
+
+!equation
+\begin{split}
+&\min_p f(u, \sigma) = \frac{1}{2}\sum^N_{i=1} \left(u_i - \tilde{u}_i\right)^2; \\
+&\text{subject to}
+\left\{\begin{array}{lr}
+\dot{u} - \vec{\nabla}\cdot(1+u)\vec{\nabla}u + \sigma u = 1 & \text{in} \, t=[0,1],\Omega=(0,1)^2\\
+u(t=0) = 0 & \\
+u(x=0) = u(y=0) = 0 & \\
+\left.\frac{\partial u}{\partial x}\right|_{x=1} = \left.\frac{\partial u}{\partial y}\right|_{y=1} = 0 \\
+\sigma \geq 0
+\end{array}\right.,
+\end{split}
+
+where $N$ is the number of measurement locations and $u_i$ and $\tilde{u}_i$ are the simulated and measured state variables (e.g. temperature and displacement fields), respectively, at location $i$. The parameter being optimized is the spatially dependent reaction rate ($\sigma$). The domain is meshed using a $16\times16$ grid of quadrilateral elements, and time is discretized using implict Euler over ten uniform time steps. The measurement data is generated by evaluating the PDE with $\sigma = e^{xy} - 1$ and sampling the resulting solution at 17 locations---shown in top left plot of \cref{fig:diff_react}---at every time step, resulting in $N=170$.
```

Comment:
instead of "eg displacement or temperature fields" is there a variable that this reaction equation applies to like chemical species or mass fraction.... I really don't know what this equation models.